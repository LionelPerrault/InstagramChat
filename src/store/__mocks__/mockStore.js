import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
