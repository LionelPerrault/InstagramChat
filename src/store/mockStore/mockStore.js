import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
