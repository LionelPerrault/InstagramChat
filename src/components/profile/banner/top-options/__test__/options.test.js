import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../store/__mocks__/mockStore'
import BannerOptions from '../options'
import MockDataElement from '../../../../../utils/__mocks__/mock-dataElement'

describe('BannerOptions Component', () => {
  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <BannerOptions toggleOptions={jest.fn()} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot and show <CopyLink/> only', () => {
    MockDataElement()
    const tree = create(
      <Provider store={mockStore}>
        <BannerOptions toggleOptions={jest.fn()} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
