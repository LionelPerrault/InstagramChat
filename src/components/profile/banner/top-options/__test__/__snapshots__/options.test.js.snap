// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerOptions Component should match snapshot 1`] = `
<ul>
  <li>
    <a
      className="pro_block"
      href="#"
      onClick={[Function]}
    >
      Block
    </a>
  </li>
  <li>
    <a
      className="pro_recommend"
      href="#"
      onClick={[Function]}
    >
      Recommend
    </a>
  </li>
  <li>
    <a
      className="add_fav"
      href="#"
      onClick={[Function]}
    >
      Add to favourites
    </a>
  </li>
  <li>
    <a
      className="p_copy_link"
      href="#"
      onClick={[Function]}
    >
      Copy profile link
    </a>
  </li>
</ul>
`;

exports[`BannerOptions Component should match snapshot and show <CopyLink/> only 1`] = `
<ul>
  <li>
    <a
      className="p_copy_link"
      href="#"
      onClick={[Function]}
    >
      Copy profile link
    </a>
  </li>
</ul>
`;
